File: CareConnect/CareConnect.Models/Requests/WorkshopInsertRequest.cs
using System;
using System.Text.Json.Serialization;

namespace CareConnect.Models.Requests
{
    public class WorkshopInsertRequest
    {
        public string Name { get; set; } = null!;

        public string Description { get; set; } = null!;

        public string WorkshopType { get; set; }

        public DateTime Date { get; set; }

        public decimal? Price { get; set; }

        public int? MaxParticipants { get; set; }

        public int? Participants { get; set; }

        public string? Notes { get; set; }

        [JsonIgnore]
        public DateTime ModifiedDate { get; set; } = DateTime.Now;
    }
}

File: UI/careconnect_admin/lib/models/requests/workshop_insert_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'workshop_insert_request.g.dart'; // Don't open or edit this

@JsonSerializable()
class WorkshopInsertRequest {
  String name;
  String description;
  String workshopType;
  DateTime startDate;
  DateTime? endDate;
  double? price;
  double? memberPrice;
  int? maxParticipants;
  int? participants;
  String? notes;

  WorkshopInsertRequest({
    required this.name,
    required this.description,
    required this.workshopType,
    required this.startDate,
    this.endDate,
    this.price,
    this.memberPrice,
    this.maxParticipants,
    this.participants,
    this.notes,
  });

  factory WorkshopInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$WorkshopInsertRequestFromJson(json);

  Map<String, dynamic> toJson() => _$WorkshopInsertRequestToJson(this);
}
